Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -0,0 +1,2769 @@
+    Args:
+        msg (str): Message to be logged.
+
+    """
+    debug_msg = f'QRadarMsg - {msg}'
+    if lock.acquire(timeout=LOCK_WAIT_TIME):
+        demisto.debug(debug_msg)
+        lock.release()
+
+
+def is_reset_triggered(handle_reset: bool = False):
+    """
+    Checks if reset of integration context have been made by the user.
+    Because fetch is long running execution, user communicates with us
+    by calling 'qradar-reset-last-run' command which sets reset flag in
+    context.
+    Using lock because calls to this function concurrent
+    Args:
+        handle_reset (bool): Whether the reset should be handled by the caller.
+
+    Returns:
+        (bool):
+        - True if reset flag was set. If 'handle_reset' is true, also resets integration context.
+        - False if lock could not be acquired until timeout.
+        - False if lock was acquired, and reset flag was not found in integration context.
+    """
+    if lock.acquire(timeout=LOCK_WAIT_TIME):
+        ctx = get_integration_context()
+        if ctx and RESET_KEY in ctx:
+            if handle_reset:
+                print_debug_msg('Reset fetch-incidents.')
+                set_integration_context({'samples': ctx.get('samples', [])})
+            lock.release()
+            return True
+        lock.release()
+    return False
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module_command(client: Client, params: Dict) -> str:
+    """
+    Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+    Args:
+        client (Client): Client to perform the API calls.
+        params (Dict): Demisto params.
+
+    Returns:
+        - (str): 'ok' if test passed
+        - raises DemistoException if something had failed the test.
+    """
+    try:
+        client.test_connection()
+        is_long_running = params.get('longRunning')
+        if is_long_running:
+            ip_enrich, asset_enrich = get_offense_enrichment(params.get('enrichment', 'IPs And Assets'))
+            get_incidents_long_running_execution(
+                client=client,
+                offenses_per_fetch=int(params.get('offenses_per_fetch', DEFAULT_OFFENSES_PER_FETCH)),
+                user_query=params.get('query', ''),
+                fetch_mode=params.get('fetch_mode', FETCH_MODE_DEFAULT_VALUE),
+                events_columns=params.get('events_columns', EVENT_COLUMNS_DEFAULT_VALUE),
+                events_limit=int(params.get('events_limit', DEFAULT_EVENTS_LIMIT)),
+                ip_enrich=ip_enrich,
+                asset_enrich=asset_enrich,
+                last_highest_id=get_integration_context().get(LAST_FETCH_KEY, 0),
+                incident_type=params.get('incident_type'),
+                mirror_direction=MIRROR_DIRECTION.get(params.get('mirror_options', DEFAULT_MIRRORING_DIRECTION))
+            )
```

Comment:
Please rethink this test.

You are fully simulating fetch-incidents with (potentially) events fetch. The main issue about this is that it might take a long time - fetching incidents with events can take > 5 minutes for a single fetch, especially on servers that have a heavy load on them.
I believe you should limit this test to a single offense and a single event to limit its run time. I'm not sure whether you're interested in waiting for the `events fetch` to complete either - if the user has the authorization to execute a search, he will also be able to retrieve its results.

Reply:
Thanks, ill change offenses_per_fetch=1 and events_limit=1